core/src/main/scala/org/http4s/Header.scala

```diff
@@ -82,7 +82,13 @@ object Header {
    * @param value String representation of the header value
    */
   final case class Raw(name: CaseInsensitiveString, override val value: String) extends Header {
-    override lazy val parsed = parser.HttpHeaderParser.parseHeader(this).getOrElse(this)
+    private[this] var _parsed: Header = null
+    final override def parsed: Header = {
+      if (_parsed == null) {
+        _parsed = parser.HttpHeaderParser.parseHeader(this).getOrElse(this)
+      }
+      parsed
```

Comment:
Sometimes, in scala, you can accidentally write tail-recursive loops and not realize it until you realize that your cpu has been turned into a hotplate.

I remember that happening to Ivan one day when I first joined the http4s project, he was hunting for the problem for hours, poor guy. 😄